app/static/src/js/headscripts/gtm.js

```diff
@@ -23,6 +23,44 @@ function add_GTM() {
     GTM_Loaded = true;
 }
 
+function push_to_datalayer(event, category_code = null, category_name = null, category_traversal = null, diagnosis_result = null) {
+    window.dataLayer = window.dataLayer || [];
+    window.dataLayer.push({
+        event: event,
+        category_code: category_code,
+        category_name: category_name,
+        category_traversal: category_traversal,
+        diagnosis_result: diagnosis_result,
+    });
+}
+
+function diagnosed(){
+    const path = window.location.pathname;
+
+    if (path.endsWith('/legal-aid-available') || path.includes('fast-tracked')) {
+        push_to_datalayer('diagnosed', window.sessionData.category_code, window.sessionData.category_name, window.sessionData.category_traversal, "INSCOPE")
+    } else if (path.endsWith('/cannot-find-your-problem')) {
+        push_to_datalayer('diagnosed', window.sessionData.category_code, window.sessionData.category_name, window.sessionData.category_traversal, "OUTOFSCOPE")
+    }
+}
+
+function diagnosed_fala() {
+    document.addEventListener('click', function (e) {
+        const link = e.target.closest('a');
+        if (link && link.href.includes('/find-a-legal-adviser')) {
+            push_to_datalayer('diagnosed',category_name=link.textContent.trim().slice(0, 50), diagnosis_result="OUTOFSCOPE")
```

Comment:
This is really awesome because it will keep working when we switch over to big FALA. You can slice up to 99 safely but leave to 50 if not needed incase Google changes rules. Also you can send the category code from the querystring e.g. /find-a-legal-adviser?category=immas